lib/mongo/server/connection_pool/queue.rb

```diff
@@ -88,7 +87,7 @@ def disconnect!
         # @since 2.0.0
         def enqueue(connection)
           mutex.synchronize do
-            queue.unshift(connection)
+            queue.unshift(connection.record_checkin!)
             resource.broadcast
           end
         end
```

Comment:
When the queue is initialized with `@queue = Array.new(min_size) { create_connection }`  will those connections have a checkin time?